crates/sqlbuiltins/src/functions/scalars.rs

```diff
@@ -0,0 +1,870 @@
+use datafusion::logical_expr::BuiltinScalarFunction;
+
+use crate::builtins::BuiltinFunction;
+// mostly using a macro here to preserve the formatting.
+// rustfmt will otherwise compact the lines.
+macro_rules! make_const {
+    (
+        $var_name:ident,
+        name => $name:expr,
+        example => $example:expr,
+        description => $description:expr
+    ) => {
+        const $var_name: (&str, &str, &str) = ($name, $example, $description);
+    };
+}
+
+make_const!(
+    ABS,
+    name => "abs",
+    example => "abs(-1)",
+    description => "Compute the absolute value of a number."
```

Comment:
In all of these excamples we say "compute" rather than return. I think return is useful because it indicates that there's output, and I might use "compute" in situations where the function was just used for side effects.